regolith/sorters.py

```diff
@@ -4,12 +4,89 @@
 
 from regolith.dates import date_to_float
 
-doc_date_key = lambda x: date_to_float(x.get("year", 1970), x.get("month", "jan"))
-doc_date_key_high = lambda x: date_to_float(x.get("year", 1970), x.get("month", "dec"))
-ene_date_key = lambda x: date_to_float(x.get("end_year", 4242), x.get("end_month", "dec"))
-category_val = lambda x: x.get("category", "<uncategorized>")
-level_val = lambda x: x.get("level", "<no-level>")
-id_key = lambda x: x.get("_id", "")
+
+def doc_date_key(x):
+    """
+    Convert a Datetime object of the doc to a float key.
+    Parameters
+    ----------
+    x
```

Comment:
let's give this a more intuitive name like `date`

then in the numpy-style docstring it would look like
```
date:  dict
  the document that is expected to contain date-like information in "year" and/or "month"

```